### PR TITLE
support bytes-like for X25519PrivateKey.from_private_bytes

### PR DESCRIPTION
**Pull request by reaperhulk**
yuck.

### CODE REVIEW

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 1]
```diff
@@ -2109,8 +2111,12 @@ def x25519_load_private_bytes(self, data):
         # contains an OCTET STRING of length 32! So the last two bytes here
         # are \x04\x20, which is an OCTET STRING of length 32.
         pkcs8_prefix = b'0.\x02\x01\x000\x05\x06\x03+en\x04"\x04 '
-        bio = self._bytes_to_bio(pkcs8_prefix + data)
-        evp_pkey = backend._lib.d2i_PrivateKey_bio(bio.bio, self._ffi.NULL)
+        with self._zeroed_bytearray(48) as ba:
+            ba[0:16] = pkcs8_prefix
+            ba[16:] = data
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: Is data's length actually checked before this?

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 2]
```diff
@@ -2197,6 +2203,23 @@ def aead_cipher_supported(self, cipher):
...
+        zero when we're done.
+        """
+        ba = bytearray(length)
+        try:
+            yield ba
+        finally:
+            # We zero the bytearray this way because at the moment we're not
+            # sure of a better way that can guarantee it overwrites the
+            # memory and doesn't just replace the underlying char *.
+            for i in range(length):
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: Please factor this into its own function and use it in the other place that we use `memmove` to zero

Comment (reaperhulk) [reply src/cryptography/hazmat/backends/openssl/backend.py]: We're doing `memmove` in the other place on a `char[]` so `memmove` is the right thing to use there.

Comment (alex) [reply src/cryptography/hazmat/backends/openssl/backend.py]: A loop also works though.

File: src/cryptography/hazmat/backends/openssl/backend.py (hunk 1 above)
Comment (reaperhulk) [reply src/cryptography/hazmat/backends/openssl/backend.py]: It was not being properly checked and #4703 adds the check (along with making x25519 and x448 consistent across their from_{public,private}_bytes methods

**Pull request merged**